core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/WindowsShareCopy.java

```diff
@@ -267,8 +267,8 @@ protected void terminated() {
       this.netBiosSharename = netBiosSharename;
       this.sourceDirectory = sourceDirectory;
       this.destinationDirectory = destinationDirectory;
-      this.bufferSize = (bufferSize < MIN_BUFFER_SIZE) || (bufferSize == null) ? MIN_BUFFER_SIZE : bufferSize;
-      this.numThreads = (numThreads < MIN_NUM_THREADS) || (numThreads == null) ? MIN_NUM_THREADS : numThreads;
+      this.bufferSize = (bufferSize == null) && (bufferSize < MIN_BUFFER_SIZE) ? MIN_BUFFER_SIZE : bufferSize;
```

Comment:
What does this means ? 
The first part of  && is checking buffersize == null and since its && if its null the right hand side will be evaluated so you are camping null < MIN_BUFFER_SIZE